src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -346,8 +346,12 @@ protected final void runLeaf(Statement statement, Description description,
 
     @Override
     public Description getDescription() {
-        Description description = Description.createSuiteDescription(getName(),
-                getRunnerAnnotations());
+        Description description =
+                Description.createSuiteDescription(
+                        getTestClass().getJavaClass(),
```

Comment:
@Gordiychuk I have seen ParentRunnner used in cases where there is no Java class (for example, a JavaScript test). There are probably other cases where `geName()` is overridden (tests in JVM-based languages perhaps).

Unfortunately, the Runner APIs have many extension points so are hard to change without breaking someone. I think what you have here is safe.